backends/graphics/surfacesdl/surfacesdl-graphics.cpp

```diff
@@ -1747,12 +1747,13 @@ void SurfaceSdlGraphicsManager::grabOverlay(void *buf, int pitch) const {
 		error("SDL_LockSurface failed: %s", SDL_GetError());
 
 	byte *src = (byte *)_overlayscreen->pixels;
-	byte *dst = (byte *)buf;
-	int h = _videoMode.overlayHeight;
+	byte *dst = (byte *)surface.getPixels();
+	int h = MIN<int>(surface.h, _videoMode.overlayHeight);
```

Comment:
This might actually mask the read bug - the _videoMode.overlayHeight and the _backbuffer pixel buffer being of different sizes and ran out-of-sync. But imo it's still a good thing to perform these sanity checks here.

Reply:
Unfortunately there's a slight regression now, since it apparently only copies N bytes instead of N words, causing a glitch on the right half of the image, e.g. when pausing a SCUMM game:

![glitch](https://user-images.githubusercontent.com/601765/123689301-5c752b80-d853-11eb-81af-caa0f6045b48.png)

There's some discussion about possible solutions on the actual commit at https://github.com/scummvm/scummvm/commit/5af11925809267eee1abd5ddf251530f6f5f5572#r52686886